### DISCUSSION TITLE
"Precondition check failed" with gmail client + oauth2

### DISCUSSION THREAD

**Discussion by jamiedavenport** [googleapis/google-api-nodejs-client#3618] [Q&A]
I've setup Google OAuth with Lucia and storing the refresh token in the database.

I've tried to create a gmail client like so, using the stored refresh token:
```ts
export async function createGmailClient(refreshToken: string) {
  const auth = new google.auth.OAuth2(clientId, clientSecret, redirectUri);
  auth.setCredentials({ refresh_token: refreshToken });

  return google.gmail({ version: "v1", auth });
}
```

And then listing messages like so:
```ts
const messages = await client.users.messages.list({
      userId: "me",
    });
```

But I'm getting a `400 - Precondition check failed.` error which is strange to see since most reports talk about this error with service accounts.

I'm assuming that I've setup the `OAuth2` client incorrectly but the error messages aren't very helpful. Any help here would be hugely appreciated.

**Comment by rnuyts** [comment]
Hi @jamiedavenport,

We have the same issue with the Ruby Gmail Client and also using OAuth authentication. 

When we try to send an email for one of our users, we have this "Precondition check failed" error. For the other users (using other Google Workspace accounts), it works perfectly.

And we don't know why it happens for this specific user...

**Comment by juslin03** [comment]
Anyone got this work ?
